### PR TITLE
Refactor demo

### PR DESCRIPTION
**Pull request by jglick**
A more manageable demo structure.

* Uses the standard Jenkins base image.
* Tags images according to the Workflow plugin release.
* Inlines the demo repository so minor changes can be made in tandem with code changes.
* Fixes plugin versions in SCM for better control of changes.
* Swaps Jenkins and Jetty ports to match the expected URLs for running Jenkins.

@reviewbybees

### CODE REVIEW

**Comment by jglick** [comment]
Would benefit from https://github.com/cloudbees/jenkins-ci.org-docker/pull/89.

File: demo/Dockerfile [hunk 1]
```diff
@@ -18,31 +15,33 @@ RUN wget -O - "http://mirrors.ibiblio.org/eclipse/jetty/$JETTY_VERSION/dist/jett
...
+RUN git config --global user.email "demo@jenkins-ci.org" && git config --global user.name "Workflow Demo" && git init && git add . && git commit -m 'demo'
+
+# Prepare local Maven repo. Note that $JENKINS_HOME is a volume so we cannot populate it now.
+RUN echo '<settings><localRepository>/usr/share/jenkins/ref/.m2/repository</localRepository></settings>' > /tmp/files/settings.xml
+RUN /usr/local/maven/bin/mvn -s /tmp/files/settings.xml install && /usr/local/maven/bin/mvn -s /tmp/files/settings.xml -f sometests -Dmaven.test.failure.ignore clean install
+
+ENV JENKINS_UC http://jenkins-updates.cloudbees.com
+COPY workflow-version.txt plugins.txt /tmp/files/
+RUN sed -i "s/@VERSION@/`cat /tmp/files/workflow-version.txt`/g" /tmp/files/plugins.txt
+RUN /usr/local/bin/plugins.sh /tmp/files/plugins.txt
```
Comment (tfennelly) [inline demo/Dockerfile]: Where is the `plugins.sh` script?

Comment (tfennelly) [inline demo/Dockerfile]: Oh maybe it's in `jenkins:1.596.2` ?

**Comment by tfennelly** [comment]
:+1:

File: demo/Dockerfile (hunk 1 above)
Comment (jglick) [inline demo/Dockerfile]: Yeah, [here](https://github.com/cloudbees/jenkins-ci.org-docker/blob/66dc63923438e70eaebd68da41bc7b9ea35ce2aa/plugins.sh).

**Pull request merged**